atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -89,48 +89,13 @@ private fun Project.applyAtomicfuCompilerPlugin() {
         plugins.apply(AtomicfuKotlinCompilerPluginInternal::class.java)
     } else {
-        // for KGP >= 1.7.20:
-        // compiler plugin for JS IR is applied via the property `kotlinx.atomicfu.enableJsIrTransformation`
-        // compiler plugin for JVM IR is applied via the property `kotlinx.atomicfu.enableJvmIrTransformation`
-        if (kotlinVersion.atLeast(1, 7, 20)) {
-            // Check that the project has `org.jetbrains.kotlin:atomicfu` in the classpath, otherwise throw an error.
-            val isAtomicfuCompilerPluginOnTheClasspath = project.buildscript.configurations.findByName("classpath")
-                ?.allDependencies?.any { it.group == "org.jetbrains.kotlin" && it.name == "atomicfu" && it.version == getKotlinPluginVersion()}
-            if (isAtomicfuCompilerPluginOnTheClasspath == false) {
-                error("You are applying `kotlinx-atomicfu` plugin of version 0.23.3 or newer.\n" +
-                        "If you wish to use this version of the plugin with Kotlin version lower than 1.9.0, you must manually include `org.jetbrains.kotlin:atomicfu:${getKotlinPluginVersion()}` in your buildscript. \n" +
-                        "Here's how you can do it: \n" +
-                        "```\n" +
-                        "buildscript {\n" +
-                        "    dependencies {\n" +
-                        "        classpath(\"org.jetbrains.kotlin:atomicfu:${getKotlinPluginVersion()}\")\n" +
-                        "    }\n" +
-                        "}\n" +
-                        "```\n" +
-                        "Alternatively, you can upgrade your Kotlin version to 1.9.0 or newer.\n"
-                )
-            }
-            plugins.apply(AtomicfuKotlinGradleSubplugin::class.java)
-            extensions.getByType(AtomicfuKotlinGradleSubplugin.AtomicfuKotlinGradleExtension::class.java).apply {
-                isJsIrTransformationEnabled = rootProject.getBooleanProperty(ENABLE_JS_IR_TRANSFORMATION)
-                isJvmIrTransformationEnabled = rootProject.getBooleanProperty(ENABLE_JVM_IR_TRANSFORMATION)
-                if (kotlinVersion.atLeast(1, 9, 20)) {
-                    // Native IR transformation is available since Kotlin 1.9.20
-                    isNativeIrTransformationEnabled = rootProject.getBooleanProperty(ENABLE_NATIVE_IR_TRANSFORMATION)
-                }
-            }
-        } else {
-            // for KGP >= 1.6.20 && KGP <= 1.7.20:
-            // compiler plugin for JS IR is applied via the property `kotlinx.atomicfu.enableIrTransformation`
-            // compiler plugin for JVM IR is not supported yet
-            if (kotlinVersion.atLeast(1, 6, 20)) {
-                if (rootProject.getBooleanProperty(ENABLE_JS_IR_TRANSFORMATION_LEGACY)) {
-                    plugins.apply(AtomicfuKotlinGradleSubplugin::class.java)
-                }
-            }
-        }
+        error("You are applying `kotlinx-atomicfu` plugin of version 0.23.3 or newer. " +
```

Comment:
Previously, in case of KGP version < `1.9.0`, I asked users to add `org.jetbrains.kotlin:atomicfu` dependency manually to the buildscript. However, this caused further problems, that already were solved in higher Kotlin versions. 

E.g. the problem below would occur if you try to apply the current kotlinx-atomicfu version with Kotlin 1.8.20 and just add the `kotlin:atomicfu:1.8.20` dependency manually (_this problem was solved in `0.23.0` and Kotlin `1.9.10`_)

```
e: Could not find ".gradle/caches/modules-2/files-2.1/org.jetbrains.kotlinx/atomicfu-linuxarm64/0.23.2/7b4270560c2723dc55b1d3c8925b7b93201c437b/atomicfu.klib" in [IdeaProjects/kotlinx-atomicfu-new/integration-testing/examples/mpp-sample, .konan/klib, .konan/kotlin-native-prebuilt-macos-aarch64-1.8.20/klib/common, .konan/kotlin-native-prebuilt-macos-aarch64-1.8.20/klib/platform/linux_arm64]
```


So, I decided to write this error message about incompatibility instead.
I've checked that the sample mpp project can be built with kotlinx-atomicfu `0.22.0` and Kotlin down to `1.6.0`, though I believe these are very marginal use cases.

Reply:
> version 0.23.3 or newer

Can we print the exact version here?

I think it should be enough to:
- mention that Kotlin version should be upgraded
- add a document (like readme) describing an alternative solution for those who can't upgrade the Kotlin for some reason

Reply:
I've added a note about downgrading atomicfu version as an alternative solution in the README. And in the message left just the recommendation to upgrade the Kotlin version.